src/safeds/ml/classical/classification/_classifier.py

```diff
@@ -230,3 +233,42 @@ def f1_score(self, validation_or_test_set: TaggedTable, positive_class: Any) ->
         if (2 * n_true_positives + n_false_positives + n_false_negatives) == 0:
             return 1.0
         return 2 * n_true_positives / (2 * n_true_positives + n_false_positives + n_false_negatives)
+
+    def roc_curve(self, validation_or_test_set: TaggedTable) -> Image:
+        """
+        Compute the classifier's roc_curve on the given data.
+
+        Parameters
+        ----------
+        validation_or_test_set : TaggedTable
+            The validation or test set.
+
+        Returns
+        -------
+        roc_curve : Table
+            The calculated roc_curve, i.e. the harmonic mean between precision and recall.
+            Return 1 if there are no positive expectations and predictions.
+        """
+        if not isinstance(validation_or_test_set, TaggedTable) and isinstance(validation_or_test_set, Table):
+            raise UntaggedTableError
+
+        expected_values = validation_or_test_set.target
+        predicted_values = self.predict(validation_or_test_set.features).target
+
+        y_expected_values = []
+        y_predicted_values = []
+        for row in expected_values:
+            y_expected_values.append(row)
+        for row in predicted_values:
+            y_predicted_values.append(row)
```

Comment:
Why do we need to copy the lists? If it's really needed, use the built-in `list` function instead.